### PR TITLE
Adding scp

### PR DESCRIPTION
**Pull request by kwilcox**

### CODE REVIEW

**Comment by msarahan** [comment]
There is precedent for having this named "scp" on conda-recipes (https://github.com/conda/conda/issues/1743), but I'd really like to somehow indicate that this is a python library, NOT the scp program that many of us are familiar with.  Can we discuss how to name this on conda-forge?

**Comment by jakirkham** [comment]
Would you like a `python-` prefix? That seems reasonable.

**Comment by kwilcox** [comment]
I understand the confusion between this and some secure copy binary. Is it worth the deviation from PyPi?  If so, `python-scp` seems reasonable.

**Comment by msarahan** [comment]
IMHO, we should have it here as python-scp, but I'm happy to be overruled on that.  The issue I pointed to shows how irate some people can be when they don't get what they expect.

However, conda-forge is more than PyPI - we have packages from more places, although it does primarily come from stuff that is shared with PyPI.  I think it is in our best interest to disambiguate this with the namespace prefix, but I also think that it will inevitably cause confusion and frustration.  Yuck.

**Comment by jakirkham** [comment]
SGTM. @ocefpaf, thoughts on naming?

**Comment by kwilcox** [comment]
If we are not trying to maintain some sort of PyPI compat, is there anything preventing me from submitting **all** python packages with the `python-` prefix to prevent future collisions?

**Comment by jakirkham** [comment]
Maybe we should be doing that honestly. The previous discussion basically resulted in us saying we would try to pick the original name, but if that failed to prefix with `python-`. ( https://github.com/conda-forge/conda-forge.github.io/issues/18 ) Though I can see a good argument that doing the prefix only sometimes makes things a little more confusing than just always having the prefix.

**Comment by kwilcox** [comment]
`python-python-dateutil` :laughing: 

I'll wait to see what @ocefpaf says about this. I'll be with him next week for a few days and we'll discuss some more. In the meantime I have this in another repo that I'll use.

**Comment by jakirkham** [comment]
> `python-python-dateutil` :laughing: 

lol, yeah, I have tended to stick with one. Though here's a fun situation. There is already [`python-coveralls`]( https://pypi.python.org/pypi/python-coveralls ) and [`coveralls`]( https://pypi.python.org/pypi/coveralls ). Maybe one would need `python-python-coveralls`? 😝

> I'll wait to see what @ocefpaf says about this.

👍

> In the meantime I have this in another repo that I'll use.

Sorry about that. 😦

**Comment by ocefpaf** [comment]
TL;DR

`python-scp`

I like to add prefix when there is no prefix already. `python-scp` sounds good, but `python-python-datetime` is a :scream_cat: 

Some sort of package name-spacing is needed! We adopt some of the default channel original names and that broke from adopting a prefix from the start. However, as @msarahan said above, we might want to start fixing these here and hopefully those change would get propagated back into defaults.

My Linux distro is adamant to add the prefix to every original python package name. Packages like `netcdf4-python` are named `python-netcdf4-python`. That sounds weird to me and I would love to have it named just `netcdf4-python`, as the original author intended.

Note that, even though I am :+1:  on adding the `python-` prefix, if we go down the prefixing road we should get ready for a lot of complains and breakages everywhere!

**Comment by jakirkham** [comment]
> Note that, even though I am :+1:  on adding the `python-` prefix, if we go down the prefixing road we should get ready for a lot of complains and breakages everywhere!

Yep, I have broken my own stack by doing this a few times, but to be honest I'm grateful that we are embracing a principled manner for solving this problem. I would happily fix breakages in my stack than break conda-forge by having weird naming all over the place.

File: recipes/python-scp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    run:
+        - python
+        - paramiko
+
+test:
+    imports:
+        - scp
+
+about:
+    home: https://github.com/jbardin/scp.py.git
```
Comment (jakirkham) [inline recipes/python-scp/meta.yaml]: Probably should drop the trailing `.git`. 😄

File: recipes/python-scp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    license: LGPL
```
Comment (jakirkham) [inline recipes/python-scp/meta.yaml]: Could we please specify the version? Looks like 2.1 maybe.

**Comment by jakirkham** [comment]
Couple of minor changes mentioned above. Once addressed this is good to merge IMHO.

**Comment by ocefpaf** [comment]
@msarahan based on the latest new this is OK as `python-scp`. (Although the confusion here is with the command `scp` and not another library. So add the prefix is not really that important. Right?)

@kwilcox if you can fix the license version I will merge this ASAP.

**Comment by jakirkham** [comment]
Though if we were to add a package for the `scp` command, I think it should get the original name. Wouldn't you agree, @ocefpaf?

**Comment by ocefpaf** [comment]
If I got conda future support for namespaces correctly I don't think that
will be a problem.
Em 14/06/2016 00:52, "jakirkham" <notifications@github.com> escreveu:
